src/main/java/filter/lexer/Lexer.java

```diff
@@ -21,6 +21,7 @@
             // These have higher priority than Symbol
             new Rule("\\d{4}-\\d{1,2}-\\d{1,2}", TokenType.DATE), // YYYY-MM?-DD?
             new Rule("[A-Za-z]+(-[A-Za-z]+)*\\s*:", TokenType.QUALIFIER),
+            new Rule("[A-Za-z0-9#][/A-Za-z0-9.'+-]+#", TokenType.COMPOUND_ID),
```

Comment:
Sure, we should do that.

So you want to allow something like `id:owner/repo#>=3`? This would be quite an elegant way to support that. In that case maybe we should do it this way. @damithc